### DISCUSSION TITLE
The state of 10.9

### DISCUSSION THREAD

**Discussion by msimic** [jellyfin/jellyfin#11051] [General] [answered]
10.9, which ought to contain some gamebreaking missing functionalities for some users, seems to be stuck for a very long time.

Is there a plan to release it soon? I don't think this would be even possible.
I see constant merged pull requests and failed builds by the day.
Scope creep is becoming a problem it seems.

Any thought from the devs?

**Comment by joshuaboniface** [comment, marked as answer]
So, there's a couple discrete points to address here, slightly out of order.

>Is there a plan to release it soon?

Yes, we do have a plan: About 2 months ago we began really pushing to "finish it up", and while we - like every release - do not have firm dates, we are getting very close to 10.9 now, and currently, our *tentative* roadmap to 10.9.0 is:

1. March 2024: Finalize CI, begin releasing weekly unstable builds for public consumption, publish a blog post about the changes and how this release (and future releases) will work: tl;dr this will be the start of the "beta" phase of 10.9.0, "betas" will be on master, and we'll cut the release when we have a weekly build that's "good" major bug wise.
2. Early-April 2024: Final push for bugfixes in Master before 10.9.0 based on the feedback from the 3-5 weeks of "beta" unstable builds.
3. Mid-April: 10.9.0 stable release.

Again all of that is subject to change depending on what happens, but we're definitely on track here. Stay tuned to our blog and forums for more details as this happens.

> I see constant merged pull requests

PRs being merged is a good thing, it means work is getting done. The majority are bugfixes.

> and failed builds by the day.

The failed builds are part of why 10.9.0 has been stuck for so long. Our existing CI and release workflow was a mess and not working any longer, meaning that releases were a massive chore - taking me in particular upwards of 40 man-hours each for major releases and 10+ man-hours each for minor releases - and thus, were something we didn't want to do often. Part of the push in the last ~2 months for 10.9.0 has been to finally fix up our CI to ensure both consistent and reproduceable(ish) builds, and improve the workflows around releases, which in turn lets us massively pick up the cadence. The plan for this is proceeding nicely, and we plan to turn on our new unstable weekly builds in 1-2 weeks (more details below).

> Scope creep is becoming a problem it seems.

The reason has been not so much scope creep, but the opposite, scaling back from an overly-ambitious scope for what "10.9" would be back when 10.8 released. On the other hand, bugs abound and we've been fixing them as time goes on. Remember, everyone here volunteers free time, often on radically different schedules, so there's often a disconnect between what certain people do versus what other people do.

> seems to be stuck for a very long time.

The reasons for this are myriad. I mentioned 2 big ones above: the broken CI and workflows, and the scheduling difficulties.

The reality is, Jellyfin is a huge sprawling project and massive areas of it have only 1 or 2 people who know them well. If one of them takes a 6 month break, gets burned out, or leaves (as happened with CI), we end up stuck. This has happened a few times in the 10.9 cycle, which is why we're at almost 2 years. The insinuation here is that "scope creep" is the reason for the delays, but it's not - it's a lack of manpower.

This is why we put out a Hacktoberfest "Call for developers" back in October 2023, which had enough success to justify it, as well as assembled a Triage team from our community moderators, in order to help  improve processes long-term and ensure issues are well-triaged and don't balloon. Coupled with the above enhancements to C, this will let us pick up the cadence a lot and avoid falling into this same trap, but it's a long-term investment that has not paid of immediately. But it's close to. There's other various pain points that contributed to the delays with this release, but they're minor and I don't want to dwell too much on them, but suffice to say things are improving and we're on-track, both for 10.9.0 and a much quicker 10.10 and beyond.

That pretty much covers my thoughts on the question as it stands today.

**Reply by ageisen2000** [reply]
Is there a list of issues that need to be tackled that could be good for new people to onboard into Jellyfin?

**Reply by Gr3q** [reply]
> Yes, we do have a plan: About 2 months ago we began really pushing to "finish it up",

My question is, is there any (scrapeable by search engines) place that I cloud inform myself about these decisions without asking questions on the repo itself?

I like to follow the state of the project, but currently I feel like every decision is done behind closed doors (which is ok) and stays there (which I have the problem with).

Note I don't ask this because I want to judge decisions or anything like that, I just feel like I'm unable to find information on what's happening to the project, including the next release.

**Reply by nielsvanvelzen** [reply]
https://jellyfin.org/posts/new-ci-new-repo

**Reply by Gr3q** [reply]
It seems I just suck at searching (also I didn't check the blog in March), thank you.

**Comment by msimic** [comment]
It is good to hear and see that something is happening and there is vision.
I wasn't aware of the huge build integration changes behind the curtains.

I am closing the discussion I opened, because the comment and the blog post by Joshua fully answers it.
- there is a plan to release in the near future
- there is a code freeze in queue
- we got to hear what are the current struggles

Hope it all goes well and keep up the good work guys.

**Comment by hjkcai** [comment]
Is there any updates about the progress of 10.9? [This blog post](https://jellyfin.org/posts/testing-10.9.0) said it would be released at Apr 26-28. @joshuaboniface 

Actually I am more interested in the changelog about 10.9. Is there a blog post about it?

**Comment by joshuaboniface** [comment]
We are still bugfixing; several major bugs were found that needed 2-3 additional weeks to fix.

We're in the final stages now but I don't want to give a firm date since its slipped a few times already. But it is imminent.

As for a changelog, the latest one (from a few weeks back now) is at https://notes.jellyfin.org/github-server-10.9.0 and https://notes.jellyfin.org/github-web-10.9.0 for each component, and a generalized summary for normal users is in https://github.com/jellyfin/jellyfin.org/pull/949
